group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -559,6 +580,7 @@ public CompletableFuture<StreamsGroupHeartbeatResult> streamsGroupHeartbeat(
         }
 
         try {
+            throwIfStreamsGroupHeartbeatRequestIsUsingUnsupportedFeatures(request);
```

Comment:
Nit: should we do this check after the "is valid" check?

Reply:
I was wondering about the order, but I think if we are using an unsupported feature, we should know this from error message, even if the request is also not well-formed.